### DISCUSSION TITLE
How to switch to system timestamps, instead of RTOS timestamps, in logging?

### DISCUSSION THREAD

**Discussion by cyberman54** [espressif/arduino-esp32#7951] [Q&A]
Is there a way to have timestamps, based on system (not RTOS) time in logs?

I found logging in ESP32-IDF is [controlled by defines](https://github.com/espressif/esp-idf/blob/master/components/log/include/esp_log.h#L401) in SDKonfig, while arduino-esp32 framework [is using RTOS timestamp](https://github.com/espressif/arduino-esp32/blob/master/tools/sdk/esp32/sdkconfig#L1189) logging.

Im am using ESP_IDF logging style (USE_ESP_IDF_LOG set).

Is there a way to override the arduino-esp32 setting and switching to system time logging?

**Comment by mrengineer7777** [comment]
Short answer: No.
`CONFIG_LOG_TIMESTAMP_SOURCE_RTOS` is used in esp_log.h.  That is at the IDF level. Modifications to `sdkconfig.h` only apply if you recompile the base IDF binaries.  For that level of control it's better to code with IDF and possibly "Arduino as IDF component".
